### PR TITLE
Texture.clone should clone orig, frame, trim Rectangles

### LINKED ISSUE #6480: PIXI.Texture.clone() doesn't clone the frame
### Expected Behavior
cloned frame 

### Current Behavior
not cloned

### Possible Solution
this.frame.clone() and if not null also clone this.orig, this.trim

cheers to Ivan

### PR DESCRIPTION
**Pull request by bigtimebuddy**
Fixes #6480

### CODE REVIEW

File: packages/core/src/textures/Texture.ts [hunk 1]
```diff
@@ -309,7 +309,13 @@ export class Texture extends EventEmitter
      */
     clone(): Texture
     {
-        return new Texture(this.baseTexture, this.frame, this.orig, this.trim, this.rotate, this.defaultAnchor);
+        return new Texture(this.baseTexture,
+            this.frame.clone(),
+            this.orig.clone(),
+            this.trim.clone(),
+            this.rotate,
+            this.defaultAnchor
```
Comment (themoonrat) [inline packages/core/src/textures/Texture.ts]: Are we ok with the defaultAnchor copied by reference?

Comment (EneaEntertainment) [reply packages/core/src/textures/Texture.ts]: I think trim could be undefined or null perhaps and needs check

Comment (bigtimebuddy) [reply packages/core/src/textures/Texture.ts]: defaultAnchor is cloned in the constructor anyways

Comment (bigtimebuddy) [reply packages/core/src/textures/Texture.ts]: I added another test check for `defaultAnchor` to make sure it isn't the same reference. Also, made `trim` optional (thanks @EneaEntertainment)

**Comment by ivanpopelyshev** [comment]
In other libs things are usually cloned in constructor, but we cant do that in case someone uses that behaviour.

**Pull request merged**